src/tasks/block/sim.rs

```diff
@@ -265,13 +172,19 @@ impl Simulator {
     /// An `Instant` representing the simulation deadline, as calculated by determining
     /// the time left in the current slot and adding that to the current timestamp in UNIX seconds.
     pub fn calculate_deadline(&self) -> Instant {
-        // Calculate the current timestamp in seconds since the UNIX epoch
-        let now = SystemTime::now();
-        let unix_seconds = now.duration_since(UNIX_EPOCH).expect("Time went backwards").as_secs();
-        // Calculate the time remaining in the current slot
-        let remaining = self.slot_calculator.calculate_timepoint_within_slot(unix_seconds);
-        //  Deadline is equal to the start of the next slot plus the time remaining in this slot
-        Instant::now() + Duration::from_secs(remaining)
+        // Get the current timepoint within the slot.
+        let timepoint = self.slot_calculator().current_timepoint_within_slot();
+
+        // We have the timepoint in seconds into the slot. To find out what's
+        // remaining, we need to subtract it from the slot duration
+        let remaining = self.slot_calculator().slot_duration() - timepoint;
+
+        // We add a 1500 ms buffer to account for sequencer stopping signing.
```

Comment:
might be redundant, but is this buffer also accounting for the block query cutoff itself? I thought quincey stopped signing 2s before the end of the slot

Reply:
ahhh good point. the builder needs to know this parameter 🤔

Reply:
I assume this will be implemented on another PR?

Reply:
can you make a linear ticket for me, im afk 😅

Reply:
already exists, we can just reassign cc @dylanlott https://linear.app/initiates/issue/ENG-992/respect-slot-timing-rules-in-the-builder-submit-task